hazelcast/src/main/java/com/hazelcast/client/impl/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -801,7 +801,7 @@ private void handleAuthResponse(ClientConnection connection, ClientAuthenticatio
                 .getConnectedServerVersion() + " Local address: " + connection.getLocalSocketAddress());
 
         if (changedCluster) {
-            clusterConnectionExecutor.execute(() -> sendStatesToCluster(newClusterId));
+            sendStatesToCluster(newClusterId);
```

Comment:
This does not guarantee that sendStates will work. If you check the method, in failure, it offloads the task  into an executor.